hypervisor/src/hypervisor/partition.rs

```diff
@@ -128,76 +140,83 @@ impl Run {
-                    Some(&base.bin),
-                    &bin,
-                    None,
-                    MsFlags::MS_RDONLY | MsFlags::MS_BIND,
-                    None,
-                )
-                .unwrap();
-
-                let dev = base.working_dir.path().join("dev");
-                std::fs::create_dir_all(&dev).unwrap();
-                // TODO bind-mount requested devices
-
-                // Mount /dev/null (for stdio::null)
-                let null = dev.join("null");
-                File::create(&null).unwrap();
-                mount::<_, _, str, str>(
-                    Some("/dev/null"),
-                    &null,
-                    None,
-                    MsFlags::MS_RDONLY | MsFlags::MS_BIND,
-                    None,
-                )
-                .unwrap();
-
-                // Mount proc
-                let proc = base.working_dir.path().join("proc");
-                std::fs::create_dir(proc.as_path()).unwrap();
-                mount::<str, _, str, str>(
-                    Some("/proc"),
-                    proc.as_path(),
-                    Some("proc"),
-                    MsFlags::empty(),
-                    None,
-                )
-                .unwrap();
-
-                //// Mount CGroup V2
-                let cgroup = base.working_dir.path().join("sys/fs/cgroup");
-                std::fs::create_dir_all(&cgroup).unwrap();
-                mount::<str, _, str, str>(
-                    None,
-                    cgroup.as_path(),
-                    Some("cgroup2"),
-                    MsFlags::empty(),
-                    None,
-                )
-                .unwrap();
+                // Mount the required mounts
+                let mounts = [
+                    // Mount working directory as tmpfs
+                    FileMountInfo {
+                        source: None,
+                        target: "".into(),
+                        fstype: Some("tmpfs".into()),
+                        flags: MsFlags::empty(),
+                        data: Some("size=500k".into()),
+                        is_dir: true,
+                    },
+                    // Mount binary
+                    FileMountInfo {
+                        source: Some(base.bin.clone()),
+                        target: "bin".into(),
+                        fstype: None,
+                        flags: MsFlags::MS_RDONLY | MsFlags::MS_BIND,
+                        data: None,
+                        is_dir: false,
+                    },
+                    // Mount /dev/null (for stdio::null)
+                    FileMountInfo {
+                        source: Some("/dev/null".into()),
+                        target: "dev/null".into(),
+                        fstype: None,
+                        flags: MsFlags::MS_RDONLY | MsFlags::MS_BIND,
+                        data: None,
+                        is_dir: false,
+                    },
```

Comment:
Maybe we should add a `Default` implementation to the `FileMountInfo` struct so that we can use the [Struct Update Syntax](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax)

```Rust
FileMountInfo {
  source: Some("/dev/null".into()),
  target: "dev/null".into()
  flags: MsFlags::MS_RDONLY | MsFlags::MS_BIND,
  ..FileMountInfo::default(),
}
```

Reply:
I'm personally somewhat against that. Most of the member fields offer no semantically correct default, with flags and data being the notable exception.